app/controllers/TEAHubController.scala

```diff
@@ -9,16 +10,33 @@ import scala.concurrent.{ExecutionContext, Future}
 /** Is a controller for the project
   *
   * @param togglService Service that requests Toggl the list of the projects */
-class TEAHubController(togglService: TogglService)(implicit executionContext: ExecutionContext) extends Controller {
+class TEAHubController(togglService: TogglService, cache: CacheApi)(implicit executionContext: ExecutionContext) extends Controller {
```

Comment:
Because response body can be huge so OkHttp doesn’t store it in memory, it reads it as a stream from network when you need it.
When you read body as a string() OkHttp downloads response body and returns it to you without keeping reference to the string, it can’t be downloaded twice without new request.

Check this issue plz => https://github.com/square/okhttp/issues/1240
Adding cache was the solution that came into my mind.